scio-core/src/main/java/com/spotify/scio/util/RemoteFileUtil.java

```diff
@@ -103,10 +107,37 @@ public Path download(URI src) {
    * @return {@link Path}s to the downloaded local files.
    */
   public List<Path> download(List<URI> srcs) {
+    return download(srcs, CONCURRENCY_LEVEL);
+  }
+
+  /**
+   * Download a batch of remote {@link URI}s in parallel, using at most numThreads to do so.
+   * `numThreads` may not be larger than the number of available processors * 4.
```

Comment:
```suggestion
   * Specifying a `numThreads` greater than number of available processors * 4 will have the same effect as specifying `numThreads` equal to the available processors * 4.
```

Or maybe 
```suggestion
   * `numThreads` should at maximum be equal to the number of available processors * 4.
```
and then actually clamp the size of the thread pool